projects/igniteui-angular/src/lib/grids/grid/grid-validation.spec.ts

```diff
@@ -0,0 +1,405 @@
+
+            grid.crudService.endEdit(true);
+            fixture.detectChanges();
+
+            GridFunctions.verifyCellValid(cell, true);
+            expect(grid.validationStatusChange.emit).toHaveBeenCalledWith(Validity.Valid);
+        });
+
+        xit('should return invalid transaction using the transaction`s getInvalidTransactionLog method', () => {
+            const grid = fixture.componentInstance.grid as IgxGridComponent;
+
+
+            let cell = grid.gridAPI.get_cell_by_visible_index(1, 1);
+            UIInteractions.simulateDoubleClickAndSelectEvent(cell.element);
+            cell.update('asd');
+            fixture.detectChanges();
+            cell = grid.gridAPI.get_cell_by_visible_index(1, 1);
+            let transactionLog = grid.validation.getInvalid();
+
+            GridFunctions.verifyCellValid(cell, false);
+            expect(transactionLog[0].state[0].valid).toBeFalse();
+
+
+            cell.editMode = true;
+            cell.update('test');
+            fixture.detectChanges();
+            cell = grid.gridAPI.get_cell_by_visible_index(1, 1);
+
+            GridFunctions.verifyCellValid(cell, true);
+            transactionLog = grid.validation.getInvalid();
+            expect(transactionLog[1].state[0].valid).toBeTrue();
+        });
+    });
+
+    describe('Custom Validation - ', () => {
+        let fixture;
+
+        beforeEach(fakeAsync(() => {
+            fixture = TestBed.createComponent(IgxGridValidationTestCustomErrorComponent);
+            fixture.detectChanges();
+        }));
+
+        it('should allow setting custom validators via template-driven configuration on the column', () => {
+            const grid = fixture.componentInstance.grid as IgxGridComponent;
+            let cell = grid.gridAPI.get_cell_by_visible_index(1, 1);
+
+            UIInteractions.simulateDoubleClickAndSelectEvent(cell.element);
+            cell.update('bob');
+            fixture.detectChanges();
+
+            cell = grid.gridAPI.get_cell_by_visible_index(1, 1);
+            //bob cannot be the name
+            GridFunctions.verifyCellValid(cell, false);
+            const erorrMessage = cell.errorTooltip.first.elementRef.nativeElement.children[0].textContent;
+            expect(erorrMessage).toEqual(' This name is forbidden. ');
+
+            cell.editMode = true;
+            cell.update('test');
+            fixture.detectChanges();
+            cell = grid.gridAPI.get_cell_by_visible_index(1, 1);
+            GridFunctions.verifyCellValid(cell, true);
+        });
+    });
+
+    describe('Trensactions integration - ', () => {
+        let fixture;
+
+        beforeEach(fakeAsync(() => {
+            fixture = TestBed.createComponent(IgxGridValidationTestBaseComponent);
+            fixture.componentInstance.batchEditing = true;
+            fixture.detectChanges();
+        }));
+
+        it('should allow setting built-in validators via template-driven configuration on the column', () => {
```

Comment:
Title does not match what is tested here. Seems validator integration with test undo/redo transaction API.